### DISCUSSION TITLE
Lose Message when delete destination Transformer

### DISCUSSION THREAD

**Discussion by IbrahimAlhaj9** [nextgenhealthcare/connect#5748] [Q&A] [answered]
Hello everyone, 
I have a channel with a File Writer destination. When I imported an existing transformer into the destination transformer and then deleted the transformer step, I lost the output message content when I deployed the channel.
![dest](https://user-images.githubusercontent.com/54806435/230615235-7225739a-2861-4af6-8509-353d73535f76.PNG)
![destination](https://user-images.githubusercontent.com/54806435/230615311-45d966fa-0232-4838-80df-6aea5fcd46c2.PNG)

and this ${message.encodedData} return ``<empty/>``

Thank you

**Comment by IbrahimAlhaj9** [comment]
Any Help with this issue?
Thanks in advance.

**Comment by IbrahimAlhaj9** [comment]
I created a pass-through step in the destination transformer as workaround to solve this issue.
![destination transformar](https://user-images.githubusercontent.com/54806435/235060805-6d9035b6-c43c-462b-9895-d6a062f5d822.PNG)

@pacmano1 I need some help figuring out why this issue has arisen. Can you assist me?
Thank you in advance.

**Reply by pacmano1** [reply]
If you "lost" the message I suspect your transformer has an outbound message template that is empty or minimal.  If JSON, you might have ```{}``` in there.   Your screenshot omits the outbound message template which is on the lower right of that screenshot had you captured the enture mirth window:

![image](https://user-images.githubusercontent.com/44065187/235217106-8a14fafd-16b6-47ad-908b-45c353cf252f.png)

**Reply by mjtrac** [reply, marked as answer]
To add to what pacmano is saying, there's no definitive output "message."  The JavaScript/ECMAScript variable msg is your inbound message data restructured (usually) as an XML tree.  But if you have an outbound message template, what you populate into that becomes the thing that is carried as your message, and it is available in JavaScript/ECMAScript as the variable tmp.  And it is usually by filling out yet another template that you indicate what you'd like to send, which is taken from an object named message, in which the final encoded form of the message at each destination is then often called the "Sent" form of the message.

**Reply by IbrahimAlhaj9** [reply]
Thank you for your help.
When I imported a transformer to the destination and deleted its steps, I also need to delete the template message to avoid a step count of zero.

Thank you.
